### PR TITLE
Feature/add umt5

### PR DESCRIPTION
**Pull request by SII-limingliu**
## Motivation

add umt5 model

## Checklist

- [ ] Please use English, otherwise it will be closed.
- [ ] The purpose of the PR, or link existing issues this PR will resolve.
- [ ] The test plan, such as providing test command.
- [ ] (Optional) The necessary documentation update.

### CODE REVIEW

File: python/sgl_jax/srt/models/umt5.py [hunk 1]
```diff
@@ -0,0 +1,1139 @@
...
+        **kwargs
+    ):
+        """
+        Forward pass with two modes:
+        1. Testing: Direct parameters (input_ids, decoder_input_ids, etc.)
+        2. SGLang: forward_batch, token_to_kv_pool, logits_metadata
+        """
+        deterministic = kwargs.get("deterministic", False)
+            
+        # Case 1: Testing mode
```
Comment (pathfinder-pf) [inline python/sgl_jax/srt/models/umt5.py]: The testing mode should be removed, and the inference mode should be used during testing.

File: python/sgl_jax/srt/models/umt5.py [hunk 2]
```diff
@@ -0,0 +1,1139 @@
...
+        
+        # Self Attention sublayer
+        self.layer0_LayerNorm = RMSNorm(
+            config.d_model, 
+            epsilon=config.layer_norm_epsilon, 
+            dtype=dtype,
+            param_dtype=dtype,
+            use_scale=True,
+        )
+        self.layer0_SelfAttention = UMT5Attention(
```
Comment (SiqiLi-Fighting) [inline python/sgl_jax/srt/models/umt5.py]: same as layer0_LayerNorm

File: python/sgl_jax/srt/models/umt5.py [hunk 3]
```diff
@@ -0,0 +1,1139 @@
...
+        values = jnp.transpose(values, (2, 0, 1))[None, :, :, :]
+        return values
+
+    def __call__(
+        self,
+        hidden_states: jax.Array,
+        mask: jax.Array | None = None,
+        deterministic: bool = False,
+        encoder_hidden_states: jax.Array | None = None,
+        encoder_mask: jax.Array | None = None,
```
Comment (SiqiLi-Fighting) [inline python/sgl_jax/srt/models/umt5.py]: these two mask will not be used when use RPA kernel
we should build custom_mask in forward_meta_data when we use sglang-jax's RPA kernel

**Review by SiqiLi-Fighting — APPROVED** [review]
